tests/java_integers.rs

```diff
@@ -0,0 +1,81 @@
+extern crate error_chain;
+extern crate jni;
+
+use error_chain::ChainedError;
+use jni::{InitArgsBuilder, JavaVM, JNIEnv};
+use jni::objects::JObject;
+use jni::objects::JValue;
+use jni::sys::JNI_VERSION_1_8;
+
+fn print_exception(env: &JNIEnv) {
+    let exception_occurred = env.exception_check().unwrap_or_else(
+        |e| panic!(format!("{:?}", e)),
+    );
+    if exception_occurred {
+        env.exception_describe().unwrap_or_else(
+            |e| panic!(format!("{:?}", e)),
+        );
+    }
+}
+
+#[test]
+fn test_java_integers() {
+    unsafe {
+        let jvm_args = InitArgsBuilder::new()
+            .version(JNI_VERSION_1_8)
+            .option("-Xcheck:jni")
+            .option("-Xdebug")
+            .build()
+            .unwrap_or_else(|e| {
+                panic!(format!("{}", e.display_chain().to_string()));
+            });
+
+        let (jvm, env) = JavaVM::new(jvm_args).unwrap_or_else(|e| {
+            panic!(format!("{}", e.display_chain().to_string()));
+        });
+
+        for int_value in -10..10 {
+            for array_length in 1..50 {
+                env.push_local_frame(0).unwrap_or_else(|e| {
+                    print_exception(&env);
+                    panic!(format!("{}", e.display_chain().to_string()));
+                });
+
+                let integer_value = env.new_object("java/lang/Integer", "(I)V", &[JValue::Int(int_value)])
+                    .map(|x| JObject::from(x))
+                    .and_then(|x| env.new_global_ref(x))
+                    .unwrap_or_else(|e| {
+                        print_exception(&env);
+                        panic!(format!("{}", e.display_chain().to_string()));
+                    });
+
+                let int_array =
+                    env.new_object_array(array_length, "java/lang/Integer", integer_value.as_obj())
+                        .map(|x| JObject::from(x))
+                        .and_then(|x| env.new_global_ref(x))
+                        .unwrap_or_else(|e| {
+                            print_exception(&env);
+                            panic!(format!("{}", e.display_chain().to_string()));
+                        });
+
+                let result = env.call_static_method(
+                    "java/util/Arrays",
+                    "binarySearch",
+                    "([Ljava/lang/Object;Ljava/lang/Object;)I",
+                    &[JValue::Object(int_array.as_obj()), JValue::Object(integer_value.as_obj())],
+                ).and_then(|x| x.i())
+                    .unwrap_or_else(|e| {
+                        print_exception(&env);
+                        panic!(format!("{}", e.display_chain().to_string()));
+                    });
+
+                assert!(0 <= result && result < array_length);
+
+                env.pop_local_frame(JObject::null()).unwrap_or_else(|e| {
```

Comment:
You might consider using [`env.with_local_frame`](https://docs.rs/jni/0.8.1/jni/struct.JNIEnv.html#method.with_local_frame) so that the lib calls `pop_…` for you.